074-additional-volumes-support.md

```diff
@@ -0,0 +1,138 @@
+# Support of additional volumes in pod
+
+We propose to enhance the Strimzi Kafka Operator by adding support for specifying additional volumes in the CRDs. This will involve modifying the Kafka CRDs (see details below) to include a `volumes` field which will allow users to define extra volumes, and a `volumeMounts` field which will allow users to define extra volume mounts.
+
+The specific usecase of ours, is that we want to set custom log4j logging for audit logs, to ensure that audit logs are *not* present in stdout, but can be picked up by a sidecar which uses OpenTelemtry or logstash to forward to the audit logs to a secure location.
+See motivation for more use-cases and see also original discussion in <https://github.com/strimzi/strimzi-kafka-operator/issues/3693>
+
+## Current situation
+
+The operator handles many configurations and operational aspects but lacks support for specifying additional volumes in the Custom Resource Definition (CRD). This limitation can restrict users from accessing log files or integrating with external storage solutions directly.
+
+## Motivation
+
+Adding support for additional volumes in the CRD will allow users to:
+
+ - Access log files for enhanced monitoring and debugging.
+ - Use additional configurations or secret management tools that require mounting extra volumes.
+ - Using for analyzing JVM issues (e.g. heap dumps)
+
+And prepare support for future use-cases such as:
+
+- Using Kafka Connect connectors to read/write files to/from the disk
+- Using it for tiered storage
+- CSI Secrets
+- Adding plugins to the container image
+
+Improve flexibility in managing Kafka deployments by allowing users to attach custom storage solutions based on their specific requirements.
+
+## Proposal
+
+### Considerations
+
+- CRD Changes: Update the Kafka CRDs to include a `volumes` and a `volumeMounts` field.
+- Validation: All additional mounted paths should be inside `/mnt` to ensure backwards compatibility for the evolution of kafka and this operator.

```

Comment:
I thought we agreed on a blacklist rather than /mnt?

Reply:
I am quite sure there was a pushback from maintainers to use `/mnt` for now, as Strimzi would like to have a lot of responsibility regarding backwards compatible with users of the operators. 

EDIT: Just double checked: https://youtu.be/_8xd_NsM2Fo?si=4qmgq-q7pI-6XvDk&t=1122

Reply:
Actually I supported @tombentley to use `/mnt`